airflow/models/dagrun.py

```diff
@@ -1158,7 +1158,7 @@ def _create_task_instances(
             # TODO[HA]: We probably need to savepoint this so we can keep the transaction alive.
             session.rollback()
 
-    def _find_missing_task_indexes(
+    def _find_missing_task_indexes_or_mark_removed(
```

Comment:
I'm not sure I can put into words why, but I don't like this function name now. It sounds like it does two *different* actions, even though it's basically "fix up if TI length has changed". And because I don't have the words, I also can't suggest a better name. Helping! :)

Reply:
"repair" maybe?

Reply:
How about `_check_for_new_or_removed_task_indexes` or `_find_new_task_indexes_and_repair_removed`?

Reply:
or simply `_repair_mapped_task_indexes`

Reply:
I like the last one, but perhaps `revise` rather than `repair`?